### PR TITLE
Remind contributors not to force-push

### PR DESCRIPTION
**Pull request by daverodgman**
## Description

A couple of hints to contributors to help make PRs more easily reviewable

## PR checklist

Please tick as appropriate and edit the reasons (e.g.: "backport: not needed because this is a new feature")

- [x] **changelog** not required
- [ ] **backport** TODO
- [x] **tests**  not required

### CODE REVIEW

**Comment by mpg** [comment]
I think we want a link to https://mbed-tls.readthedocs.io/en/latest/reviews/review-for-contributors/ as well as it explains this in more detail.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -18,3 +18,10 @@ Please tick as appropriate and edit the reasons (e.g.: "backport: not needed bec
 
 Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
 checklist for PR contributors.
+
+Help make review efficient:
+* Multiple simple commits
+  - please structure your PR into a series of small commits, each of which does one thing
+* No force-push
+  - please do not force-push to update your PR - just add new commit(s)
+  - use `git merge` (or the GitHub web interface) to resolve conflicts - not `git rebase`
```
Comment (ronald-cron-arm) [inline .github/pull_request_template.md:27]: I think that our development process says the opposite, in case of conflict we prefer a rebase.

**Review by tom-cosgrove-arm — APPROVED** [review]

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -18,3 +18,10 @@ Please tick as appropriate and edit the reasons (e.g.: "backport: not needed bec
 
 Please refer to the [contributing guidelines](https://github.com/Mbed-TLS/mbedtls/blob/development/CONTRIBUTING.md), especially the
 checklist for PR contributors.
+
+Help make review efficient:
+* Multiple simple commits
+  - please structure your PR into a series of small commits, each of which does one thing
+* No force-push
```
Comment (mpg) [inline .github/pull_request_template.md:25]: I wouldn't go that far. I'd say the _normal_ way of updating if by adding new commits, and force-pushing is the exception and should be agreed with reviewers. (In particular, I think in our usage, for resolving conflicts, rebasing is more common than merging.)

File: .github/pull_request_template.md (hunk 1 above)
Comment (gilles-peskine-arm) [reply .github/pull_request_template.md:27]: I'd say that it depends: early in the review process, a merge can be annoying. Especially if that's during the first review, and the second reviewer has to review some coding, a merge and more coding. At the other extreme, when both reviewers have approved but there's a conflict, resolving through merging can be easier to handle sometimes. However `git range-diff` tends to make rebases easy to review. So I would tend to favor rebasing as the default strategy, without prohibiting merges.

Comment (gilles-peskine-arm) [inline .github/pull_request_template.md:27]: Another annoying thing external contributors sometimes to is to merge even when there are no conflicts. I guess they're influenced by some projects that want pull requests to always be a fast-forward from the target branch. I don't know if it's prevalent enough that we need to explicitly admonish against it, but if we mention merging, we should make it clear that we only want merging if it's really necessary.

Comment (mpg) [reply .github/pull_request_template.md:27]: That's the advantage of a link to https://mbed-tls.readthedocs.io/en/latest/reviews/review-for-contributors/  - it has rooms for more details. I think here we should stick with the simple rule that the _normal_ way to rework a PR is by adding commits, and refer to the extended version for more details.

Comment (daverodgman) [reply .github/pull_request_template.md:27]: OK, I'll remove that line and link to the longer document where we can present more details

File: .github/pull_request_template.md (hunk 2 above)
Comment (daverodgman) [reply .github/pull_request_template.md]: I've very slightly softened this suggestion. But in general, I think we have difficulties from unwanted force-pushing far more than the opposite. Plus, it's easy as a reviewer to say "please force-push to tidy up these commits" - but going the other direction doesn't really work.

So, in the interest of keeping this message simple enough that casual contributors are likely to follow it, I think we should just advise against force-pushing.

File: .github/pull_request_template.md (hunk 1 above)
Comment (daverodgman) [reply .github/pull_request_template.md]: Removed this line - we can cover this in more detail in RTD.

**Review by tom-cosgrove-arm — APPROVED** [review]

**Review by mpg — APPROVED** [review]
LGTM, thanks!

**Comment by mpg** [comment]
Not sure a backport is needed: IIRC github will just pick the template from the default branch. OTOH, making a backport might be faster than making sure it's not needed ^^

**Comment by daverodgman** [comment]
> Not sure a backport is needed: IIRC github will just pick the template from the default branch. OTOH, making a backport might be faster than making sure it's not needed ^^

We do have a template in 2.28 - I assume that it is needed. I'll merge this and if it doesn't show up in 2.28, I'll make a backport.

**Comment by gilles-peskine-arm** [comment]
Templates only have an effect in the default branch. The reason 2.28 has templates is that it used to be the default branch and we never removed the template files. So there's no point in backporting.

**Pull request merged**